linode/manager
# refactor: [M3-6694] - Convert <App /> to functional component

## Description 📝
Converts the `App` class component to a functional component, using hooks instead of containers where possible. This is useful for the RQ Events work since that query is awkward to deploy as a container.

## Major Changes 🔄
- Make `App` a functional component

## How to test 🧪
1. Ensure the Cloud Manager loads and works normally.
2. Ensure `<TheApplicationIsOneFire />` component is displayed when the app crashes (you can add a random `throw` statement to trigger a crash).
3. Ensure NewRelic and analytics scripts load and work as intended.
4. Ensure keyboard shortcuts to change theme/open go-to menu work.
5. Ensure event handlers work (e.g., the linodes cache is invalidated when an update event comes in)
6. Unit tests: `yarn test`
7. E2E tests: `yarn cy:e2e`
